cfme/tests/infrastructure/test_individual_host_creds.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.flash as flash
+import pytest
+import random
+import utils.conf as conf
+
+from cfme.infrastructure.host import credential_form
+from cfme.infrastructure.provider import details_page
+from cfme.web_ui import fill
+from utils import testgen
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(metafunc, ['virtualcenter', 'rhevm'])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+# Tests to automate BZ 1278904
+def test_single_host_bad_creds(setup_provider, provider):
+    """  Tests multiple host credentialing  with good credentials """
+
+    sel.force_navigate('infrastructure_provider', context={'provider': provider})
+    sel.click(details_page.infoblock.element("Relationships", "Hosts"))
+
+    host_name = random.choice(provider.get_yaml_data()["hosts"])
+    pytest.sel.force_navigate('infrastructure_host_edit', context={
```

Comment:
I'd actually like to see ```force_navigate``` in a separate module the rest of the ```sel``` stuff for separation purposes. ```sel``` has too many ***plumbing*** functions which should be used inside the framework, but not inside the tests. What you think to that @mfalesni, if we start splitting away ***porcelain*** and ***plumbing***. We already know that pytest_selenium is not a fixture and is in the wrong place.